File: java/placedOrdersStatistics/src/entities/Order.java
package entities;

import javax.xml.crypto.Data;
import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

public class Order {
    private List<Item> items;
    private String clientName;
    private String contact;
    private String shippingAdress;
    private Double grandTotal;
    private LocalDate requestDate;

    public Order(List<Item> items, String clientName, String contact, String shippingAdress, Double grandTotal, LocalDate requestDateHour) {
        this.items = items;
        this.clientName = clientName;
        this.contact = contact;
        this.shippingAdress = shippingAdress;
        this.grandTotal = grandTotal;
        this.requestDate = requestDateHour;
    }

    public List<Item> getItems() {
        return items;
    }

    public void setItems(List<Item> items) {
        this.items = items;
    }

    public String getClientName() {
        return clientName;
    }

    public void setClientName(String clientName) {
        this.clientName = clientName;
    }

    public String getContact() {
        return contact;
    }

    public void setContact(String contact) {
        this.contact = contact;
    }

    public String getShippingAdress() {
        return shippingAdress;
    }

    public void setShippingAdress(String shippingAdress) {
        this.shippingAdress = shippingAdress;
    }

    public Double getGrandTotal() {
        return grandTotal;
    }

    public void setGrandTotal(Double grandTotal) {
        this.grandTotal = grandTotal;
    }

    public LocalDate getRequestDate() {
        return requestDate;
    }

    public void setRequestDate(LocalDate requestDateHour) {
        this.requestDate = requestDateHour;
    }

    @Override
    public String toString() {
        return "Order{" +
                ", clientName='" + clientName + '\'' +
                ", contact='" + contact + '\'' +
                ", shippingAdress='" + shippingAdress + '\'' +
                ", grandTotal=" + grandTotal +
                ", requestDateHour=" + requestDate +
                "items=" + items +
                '}';
    }
}

File: scala/src/main/scala/entity/Order.scala
package entity

import java.time.{LocalDate, LocalDateTime}
import scala.collection.mutable.ListBuffer

class Order (
              private var id: Long,
              private var item: List[Int],
              private var clientName: String,
              private var contact: String,
              private var shippingAddress: String,
              private var grandTotal: Double,
              private var requestDate: String,
            ){

  def getId: Long = id
  def getItem: List[Int] = item

  def setItem(newItem: List[Int]): Unit = item = newItem

  def getClientName: String = clientName

  def setClientName(newClientName: String): Unit = {
    clientName = newClientName
  }

  def getContact: String = contact

  def setContact(newContact: String): Unit = {
    contact = newContact
  }

  def getShippingAddress: String = shippingAddress

  def setShippingAddress(newShippingAddress: String): Unit = {
    shippingAddress = newShippingAddress
  }

  def getGrandTotal: Double = grandTotal

  def setGrandTotal(newGrandTotal: Double): Unit = {
    grandTotal = newGrandTotal
  }

  def getRequestDate: String = requestDate

  def setRequestDate(newRequestDate: String): Unit = requestDate = newRequestDate

  override def toString: String = {
    s"VALUES ($clientName, $contact, $shippingAddress, $grandTotal, $requestDate) \n"
  }

  def printarOrders: Unit = println(s"('$clientName', '$contact', '$shippingAddress', $grandTotal, '$requestDate'),")


  def printarOrderItem: Unit = for (i <- item) println(s"($id, $i),")




}
